### PR TITLE
React: Sprint Challenge

### PR DESCRIPTION
**Pull request by ogwurujohnson**

### CODE REVIEW

File: starwars/src/App.js [hunk 1]
```diff
@@ -22,6 +27,24 @@ class App extends Component {
...
+
+        // same as next logic
+        if (data.previous) {
+          this.setState({ prev: true });
+          this.setState({ prevLink: data.previous });
+        } else {
+          this.setState({ prev: false });
+          this.setState({ prevLink: '' });
+        }
+
```
Comment (mattshardman) [inline starwars/src/App.js]: this is really nice

File: starwars/src/App.js [hunk 2]
```diff
@@ -33,6 +56,7 @@ class App extends Component {
     return (
       <div className="App">
         <h1 className="Header">React Wars</h1>
+        <AllCharacters next={this.state.next} prev={this.state.prev} nextLink={this.state.nextLink} prevLink={this.state.prevLink} getCharacters={this.getCharacters} characters={this.state.starwarsChars} />
```
Comment (mattshardman) [inline starwars/src/App.js]: if you are passing multiple pieces of state it is possible to use the spread operator like:

```{...this.state}```

File: starwars/src/components/AllCharacters.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+AllCharacters.propType = {
+  characters: PropTypes.array,
+  prevLink: PropTypes.string,
+  nextLink: PropTypes.string,
+  getCharacters: PropTypes.func,
+  next: PropTypes.bool,
+  prev: PropTypes.bool,
+}
```
Comment (mattshardman) [inline starwars/src/components/AllCharacters.js]: Nice job using prop types, I believeeee array has be deprecated and you are supposed to use ```
arrayOf()``` now

File: starwars/src/components/singleCharacter.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          {character.name}
+        </label>
+        <div className="about">
+          <div className="personal">
+            <p>BirthYear: {character.birth_year}</p>
+            <p>Eye-color: {character.eye_color}</p>
+            <p>Gender: {character.gender}</p>
+            <p>Hair-color: {character.hair_color}</p>
+            <p>Height: {character.height}</p>
+          </div>
```
Comment (mattshardman) [inline starwars/src/components/singleCharacter.js]: You could potentially use a .map() here ... but might be overkill

File: starwars/src/components/AllCharacters.js (hunk 3 above)
Comment (ogwurujohnson) [reply starwars/src/components/AllCharacters.js]: Fixed and pushed

Comment (ogwurujohnson) [reply starwars/src/components/AllCharacters.js]: Thanks

File: starwars/src/App.js (hunk 2 above)
Comment (ogwurujohnson) [reply starwars/src/App.js]: Tried it out and it works, I have pushed a Fix for it

File: starwars/src/App.js (hunk 1 above)
Comment (ogwurujohnson) [reply starwars/src/App.js]: Thanks

**Comment by mattshardman** [comment]
Great job on this project Johnson!

- [x] you have successfully fetched data from the api and rendered it to the screen as specified (MVP)
- [x] Great answers to the questions - really concise (MVP)
- [x] your git flow is excellent (as always!)
- [x] your code is clean and easy to read
- [x] you have introduced pagination (stretch)
- [x] nice use of de-structuring

**Possible things to work on**
- [x] There are a couple of notes in your code but nothing major!

Overall fantastic job you have easily met the MVP and completed the stretch goals, you clearly have a good understanding of react basics. Have a great weekend!

**Pull request merged**